rust/lancedb/src/remote/table.rs

```diff
@@ -206,6 +206,11 @@ impl<S: HttpSend> RemoteTable<S> {
     ) -> Result<Vec<serde_json::Value>> {
         Self::apply_query_params(&mut body, &query.base)?;
 
+        // Fill default top_k for vector queries
+        if query.base.limit.is_none() {
+            body["k"] = serde_json::Value::Number(serde_json::Number::from(DEFAULT_TOP_K));
+        }
+
```

Comment:
This is only applied to remote queries, not to local lanceDB ones.

Instead, I think the appropriate place to apply the default limit is when either `nearest_to()` or `full_text_search()` is called. That should capture both local and remote queries.